src/hooks/useJournalData.ts

```diff
@@ -230,7 +230,10 @@ const useJournalData = (
 
     useEffect(() => {
         void (async () => {
-            if (journalName && journalClient && !loading && !data) {
+            if (
+                (journalName && journalClient && !loading && !data) ||
+                refreshing
+            ) {
```

Comment:
Using refresh as a boolean and not a value allows us more control to know when things are in the process of being refreshed